### ISSUE TITLE
[iOS] Page transition leaves a black screen on device view mode chage

### ISSUE THREAD

**Issue by Nagendhiran** [Telerik-Verified-Plugins/NativePageTransitions#47]
Trying to use this library for some understanding purpose. When I tried to do slide left transition on portrait mode view to landscape view, I'm seeing the black screen with cropped html content. 

![nativescrolling-ios8](https://cloud.githubusercontent.com/assets/2039564/7258371/b0dc1b28-e879-11e4-89f8-faf10c26843f.jpg)

Its quite reproducible in actual device itself. Is there anything missing in configurations or code.? I assume its bug itself as i am using the default options.

I'm currently having the Cordova 4.3.0, ios 3.8.0 and nativepagetransition 0.2.14.

**Comment by EddyVerbruggen** [comment]
Hi, what exactly do you mean by "slide left transition on portrait mode view to landscape view"?

If you have a test project I'd be happy to try it.

**Comment by Nagendhiran** [comment]
I have uploaded one simple demo app for your reference. Please give a try with your ios simulator and let me know what you inputs if any.

https://github.com/Nagendhiran/nativetransbug

**Comment by EddyVerbruggen** [comment]
Ah, so you're using the screenorientation plugin to force landscape.

What should the behavior be like when sliding to the left in portrait mode and ending the same transition in landscape mode? I don't think you should do a transition at all in this case..

**Comment by Nagendhiran** [comment]
Hah.. I don't think so. Couple of other questions related to your explanation.

1. If you look closely the demo app, the plugin takes the landscape view screenshot properly and does the transition. But post transition, the screen views are locked with different width and heights instead of screenshot height/width.

2. How does the same portrait mode to landscape mode transition happening successfully in android version of this plugin? I know the code bases are from different technologies, but android code does this better. why?

3. Why shouldn't we use such behaviour in our app? Is there any constraints on this plugin?

**Comment by EddyVerbruggen** [comment]
I'm not saying it can't be fixed but I recommend for the time being not to make a forced portrait-landscape switch while also using a native transition. It's just that this plugin hasn't been tested with the screenorientation plugin you're also using. I hope you can appreciate the fact that there are thousands of plugins out there which can't possibly all be tested to work with all others. If serious cross-plugin issues popup we will of course make an effort to try to find a solution.

**Comment by Nagendhiran** [comment]
Yes got you. First of all, thanks for bring us such a nice plugin to get used to. I really like this plugin, thats what making me to argue and fix such an issue.

I totally agree on your point, may be you can consider this as valid issue, and fix that when your time permits.  As I am not from xcode technologies, which makes me to wait on this. If at all i fixed this with my local version, I will do a pull request and share the fix.

**Comment by EddyVerbruggen** [comment]
Your help is much appriated. I'll leave this open as we intend to look at the screenorientation plugin in the near future. Perhaps we'll be able to find a solution for your issue.
